packages/integration-tests/src/test-project-setup/passwordless_auth.ts

```diff
@@ -73,20 +78,50 @@ class PasswordlessAuthTestProject extends TestProjectBase {
     super(name, projectDirPath, projectAmplifyDirPath, cfnClient);
   }
 
+  /**
+   * @inheritdoc
+   */
   override async assertPostDeployment(
     backendId: BackendIdentifier
   ): Promise<void> {
     await super.assertPostDeployment(backendId);
+    await this.assertPasswordlessAuth(backendId);
+  }
+
+  /**
+   * creates assertions for passwordless auth
+   */
+  private async assertPasswordlessAuth(
+    backendId: BackendIdentifier
+  ): Promise<void> {
     const userPoolClientId = (
       await this.resourceFinder.find(backendId, 'AWS::Cognito::UserPoolClient')
     )[0];
     const userPoolId = (
       await this.resourceFinder.find(backendId, 'AWS::Cognito::UserPool')
     )[0];
 
-    assert.notEqual(userPoolClientId, NaN);
-    assert.notEqual(userPoolId, NaN);
+    assert.ok(userPoolClientId);
+    assert.ok(userPoolId);
+
+    const passwordlessAuthAssertions = [
+      new SignInWithMagicLink(
+        this.cognitoIdentityProviderClient,
+        userPoolId,
+        userPoolClientId,
+        getUsername()
+      ).run(),
+      new SignInWithMagicLinkWithUnverifiedAttribute(
+        this.cognitoIdentityProviderClient,
+        userPoolId,
+        userPoolClientId,
+        getUsername()
+      ).run(),
+    ];
 
-    // TODO: define tests assertions bellow
+    const assertions = await Promise.allSettled(passwordlessAuthAssertions);
+    assertions.forEach((assertion) => {
+      assert.equal(assertion.status, 'fulfilled');
+    });
```

Comment:
does it print error somewhere if not fulfilled ?

Reply:
I added a `console.error` to print the actual error